tracer/src/Datadog.Trace/Propagators/PropagationContext.cs

```diff
@@ -13,10 +16,20 @@ internal readonly struct PropagationContext
 
     public readonly Baggage? Baggage;
 
+    public readonly IEnumerable<SpanLink>? Links;
+
     public PropagationContext(SpanContext? spanContext, Baggage? baggage)
     {
         SpanContext = spanContext;
         Baggage = baggage;
+        Links = Enumerable.Empty<SpanLink>();
+    }
+
+    public PropagationContext(SpanContext? spanContext, Baggage? baggage, IEnumerable<SpanLink>? extractionSpanLinks)
+    {
+        SpanContext = spanContext;
+        Baggage = baggage;
+        Links = extractionSpanLinks ?? Enumerable.Empty<SpanLink>();
     }
 
     public bool IsEmpty => SpanContext is null && Baggage is null;
```

Comment:
Should `IsEmpty` be updated?

Reply:
It looks like the only place we use this is in the SpanContextPropagator.Inject method where we say `if (context.IsEmpty) { return; }`

In the case of the injection operation, we should only inject when SpanContext or Baggage is populated. Should I just rename this to a bool property like `ShouldInject` or `HasContextToInject` and then leave the logic unchanged?

Reply:
I've pre-empted your feedback a little and pushed a change that removes the property altogether in ff3155274b3e14423fa06630a61256f9c692649b . This way, the conditions for the Inject operation are more clear. What do you think?

Reply:
Looks good, thanks!